src/config-manager.ts

```diff
@@ -203,9 +243,14 @@ if (!config.stripe?.secret_key) {
 
 if (!config.datastore.signature_secret) {
 	LOG_ERROR('Datastore signature secret key is not set. Set the PN_ACT_CONFIG_DATASTORE_SIGNATURE_SECRET environment variable');
-	process.exit(0);
+	configValid = false;
 }
 if (config.datastore.signature_secret.length !== 32 || !hexadecimalStringRegex.test(config.datastore.signature_secret)) {
 	LOG_ERROR('Datastore signature secret key must be a 32-character hexadecimal string.');
-	process.exit(0);
+	configValid = false;
 }
```

Comment:
This isn't your fault, but since we're here. There's a missing newline between these checks, and this can (should?) be swapped into a warning instead. I can't think of any game servers which still use this, and it will be removed in the future. The deployed server doesn't even have this key set, it's hot patched to disable the `process.exit` to stop it from crashing (dirty fix I did today before leaving for an appointment)